### DISCUSSION TITLE
How to exclude multiple status values when listing issues?

### DISCUSSION THREAD

**Discussion by anatoly-scherbakov** [ankitpokhrel/jira-cli#159] [Q&A] [answered]
## Context

All of the following applies to:
```
(Version="0.0.0", GitCommit="6c1e78e4a3b57f50ffb3e2ab3c79faaf2117b9db", GoVersion="go1.17", BuildDate="2021-09-05"", Compiler="gc", Platform="linux/amd64")
```

```shell
jira issue list --status ~Done
```

will show issues that are in any other status than `Done`. What if we want to exclude more than one status?

## Comma separation

```shell
jira issue list --status ~Done,Closed
```
will return an error:

```
⠧ Fetching issues... 
Error:
  - The value 'Done,Closed' does not exist for the field 'status'.

jira: Received unexpected response '400 Bad Request'.
Please check the parameters you supplied and try again.
```
(also, it will apparently change terminal settings: the blinking cursor is no longer seen in the GNOME terminal emulator unless you execute `tput reset`).

## Multiple values

```shell
jira issue list --status ~Done --status ~Closed
```
will not return an error but many (or all?) of the issues listed will have status `Done` — which seemingly means that the latter incarnation of `--status` option overrides the previous one. Probably there should be an error if this behavior is not supported and there should be only one copy of an option.

## Summary

Is there a way to exclude multiple status values when filtering?

**Comment by ankitpokhrel** [comment, marked as answer]
The status option [only accepts a single value](https://github.com/ankitpokhrel/jira-cli/blob/71bae6b39dfe9aedb54d52cb984954f1fe29707d/internal/cmd/issue/list/list.go#L126) at the moment. However, you can use a raw query to achieve what you want.

```sh
// All issues except the one in `Done` and  `Open`
$ jira issue list -q "status not in (Done, Open)"
```

TBH, I think it would be nice if `status` would accept multiple status. I am open for discussion (and a PR :) ) on that part.

**Comment by ankitpokhrel** [comment]
> (also, it will apparently change terminal settings: the blinking cursor is no longer seen in the GNOME terminal emulator unless you execute tput reset).

Regarding this, could you please open a new discussion/issue with more details.

**Reply by anatoly-scherbakov** [reply]
Done! https://github.com/ankitpokhrel/jira-cli/issues/162
